src/machine/machine_atmega328p.go

```diff
@@ -467,3 +472,93 @@ var SPI0 = SPI{
 	sdo:  PB3,
 	sdi:  PB4,
 	cs:   PB2}
+
+// Pin Change Interrupts
+var (
+	pinCallbacks [3][8]func(Pin)
+	interrupts   [3]*interrupt.Interrupt
+	previous     [3]uint8
+)
+
+func handlePCINTInterrupts(intr uint8, port *volatile.Register8) {
+	current := port.Get()
+	for i := uint8(0); i < 8; i++ {
+		if (current>>i)&0x01 == 0x01 && pinCallbacks[intr][i] != nil {
+			pinCallbacks[intr][i](Pin(i * (intr + 1)))
+		}
+	}
+	previous[intr] = current
+}
+
+func handlePCINT0Interrupts(intr interrupt.Interrupt) {
+	handlePCINTInterrupts(avr.IRQ_PCINT0-avr.IRQ_PCINT0, avr.PINB)
+}
+
+func handlePCINT1Interrupts(intr interrupt.Interrupt) {
+	handlePCINTInterrupts(avr.IRQ_PCINT1-avr.IRQ_PCINT0, avr.PINC)
+}
+
+func handlePCINT2Interrupts(intr interrupt.Interrupt) {
+	handlePCINTInterrupts(avr.IRQ_PCINT2-avr.IRQ_PCINT0, avr.PIND)
+}
+
+func SetInterrupt(pin Pin, callback func(Pin)) (err error) {
```

Comment:
Actually, I somehow missed this but this doesn't follow the existing API. Can you update the API to follow the standard API? See https://tinygo.org/docs/reference/machine/

```suggestion
func (pin Pin) SetInterrupt(change PinChange, callback func(Pin)) (err error) {
```

Reply:
Hm, sure, I should follow the documented standard.
But I pretty sure I copied this from one of the other architectures tho, so I think there's some non-conformity elsewhere too.
...or I just messed up.

Reply:
Aah, well part of it, at least is that the atmega328p doesn't distinguish rising v falling edges on these Port Pin change interrupts, so 'PinChange' is basically meaningless.
But that's OK

Reply:
> But I pretty sure I copied this from one of the other architectures tho

I grepped through the source code and can't find a `SetInterrupt` that is not a member of a `Pin` (or `CAN`) object.

Reply:
> at least is that the atmega328p doesn't distinguish rising v falling edges on these Port Pin change interrupts, so 'PinChange' is basically meaningless.

Right. In that case, I think only `PinToggle` should be defined (leaving `PinRising` and `PinFalling` undefined will make sure programs fail to compile when used). You could perhaps also check it in `SetInterrupt` but if only `PinToggle` exists it shouldn't really be necessary.